src/state.rs

```diff
@@ -132,6 +135,16 @@ impl State {
         self.intern_fold(Symbol::root(keyword).into(), path, create_unknown_packges)
     }
 
+    /// Call `intern_path` with the symbol's path and keyword flag
+    #[inline]
+    pub fn intern_symbol(
+        &mut self,
+        symbol: &Symbol,
```

Comment:
This is okay, although it points to a weirdness — which is that symbols should *usually* be interned. So it's a bit funny to pass a *symbol* to a function meant to intern it. We should try to work toward a world where the default is for symbols to be interned.

Reply:
This is because of our API to create symbols from strings like `".lurk.foo"` out of thin air. It was removed on the mega pr but was reintroduced later on with the help of the symbol parser